test/benchmark_http_client_test.cc

```diff
@@ -126,12 +152,17 @@ class BenchmarkClientHttpTest : public Test {
     decoders_.clear();
     dispatcher_->run(Envoy::Event::Dispatcher::RunType::Block);
     EXPECT_EQ(0, inflight_response_count);
+    // If we have no expectations, then we don't test.
+    if (header_expectations != nullptr) {
+      EXPECT_THAT((*header_expectations), UnorderedElementsAreArray(called_headers));
```

Comment:
Not sure if applicable here, but it looks like Envoy does have test helpers these days in this area, e.g. see `IsSupersetOfHeaders` over at  https://github.com/envoyproxy/envoy/blob/6dabdb393e695928f7e6777445d722fe2f625413/test/extensions/filters/http/cache/cache_filter_test.cc#L97

Reply:
This seems to be very useful. Thanks otto. For now, I think that it's fine to just save and compare the paths for now. We can change to using the superset conditioning when we want to extend this further.